torch/_inductor/lookup_table/README.md

```diff
@@ -0,0 +1,153 @@
+
+### Table Not Set or Inactive
+- Standard default behavior - generates choices using heuristics and max-autotune settings
+
+## Supported Operations
+
+Currently supports: `mm`, `addmm`, `bmm`, `mm_plus_mm`, `scaled_mm` operations with
+- Triton
+- ATEN
+- DecomposeK
+
+## Table Format
+
+The table is a dictionary with keys in the format:
+```
+"input_key+op_name"
+```
+
+Where:
+- `input_key`: Generated from `KernelInputs.key` property, represents tensor shapes/dtypes/strides
+- `op_name`: Operation name (`"mm"`, `"addmm"`, etc.)
+
+Each value is a list of configuration dictionaries containing:
+- `template_id`: Template identifier (`"triton:mm"`, `"triton::mm_persistent_tma"`, `"decompose_k"`, etc.)
+- Template-specific parameters (`BLOCK_M`, `BLOCK_N`, `BLOCK_K`, `num_warps`, etc.)
+
+### Example Table
+
+This is an example table for a single input showing two configurations
+
+```python
+table = {
+  "((torch.float16, [128, 256], [0, 1]), (torch.float16, [128, 64], [64, 1]), (torch.float16, [64, 256], [256, 1]))+alpha=1&beta=1+addmm": [
+    {
+      "template_id": "triton::mm",
+      "EVEN_K": true,
+      "USE_FAST_ACCUM": false,
+      "ACC_TYPE": "tl.float32",
+      "num_stages": 2,
+      "num_warps": 4,
+      "BLOCK_M": 32,
+      "BLOCK_N": 32,
+      "BLOCK_K": 64,
+      "hint_override": null,
+      "GROUP_M": 8,
+      "template_hash": "0717af5834e39dcca7ea817f896b8d85b4886422da7a3ab5f6911b4cfe568896"
+    },
+    {
+      "template_id": "aten::bias_addmm"
+    },
+  ]
+}
+```
+
+## Source Hashing Safety
+
+The lookup table system includes source hashing to prevent using stale configurations when template code changes.
+
+### Configuration
+- **Enabled by default**: `torch._inductor.config.lookup_table.check_src_hash = True`
+- **Optional field**: Add `"template_hash"` to table entries for enhanced safety
+
+### Behavior
+When source hash checking is enabled:
+- Template configurations with `"template_hash"` fields are validated against current template source hashes
+- Mismatched hashes indicate the template code has changed since the configuration was created
+- Stale configurations are automatically filtered out with a warning message
+- Configurations without hash fields are preserved for backward compatibility or if the user wants to fly looser
+
+### Example with Template Hash
+```python
+{
+  "template_id": "triton::mm",
+  "BLOCK_M": 32,
+  "BLOCK_N": 32,
+  "BLOCK_K": 16,
+  "template_hash": "0717af5834e39dcca7ea817f896b8d85b4886422da7a3ab5f6911b4cfe568896"
+}
+```
+
+## Performance Impact
+
+- **Lookup Hit**: Eliminates heuristic choice generation and autotuning overhead (if a single choice)
+- **Lookup Miss**: Default behavior, including heuristic choice generation and autotuning
+- **Memory**: Table stored in memory, minimal overhead for key generation and lookup
+
+## Implementation Details
+
+### Key Generation
+- Device key: Uses `torch.cuda.get_device_properties().gcnArchName` (e.g., "NVIDIA H100")
```

Comment:
I guess this is the key specification part. I think this is fine if it's a bit underspecified, and subject to change.